src/test/java/de/hpi/bpt/argos/api/EntityEndpointTest.java

```diff
@@ -0,0 +1,273 @@
+		assertEquals(testEntity.getName(), entity.get("Name").getAsString());
+		assertEquals(testEntity.getStatus(), entity.get("Status").getAsString());
+		assertEquals(attributeNames.size(), entityAttributes.size());
+	}
+
+	@Test
+	public void testGetChildEntities_InvalidParentId_Success() {
+		List<Object> attributeNames = new ArrayList<>();
+
+		for (int i = 0; i < 2; i++) {
+			attributeNames.add(testEntityTypeAttributes.get(i).getName());
+		}
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getChildEntitiesUri(testEntity.getParentId() - 1, testEntity.getTypeId(), attributeNames));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		JsonArray childEntities = jsonParser.parse(request.getResponse()).getAsJsonArray();
+		assertEquals(0, childEntities.size());
+	}
+
+	@Test
+	public void testGetChildEntities_InvalidEntityTypeId_BadRequest() {
+		List<Object> attributeNames = new ArrayList<>();
+
+		for (int i = 0; i < 2; i++) {
+			attributeNames.add(testEntityTypeAttributes.get(i).getName());
+		}
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getChildEntitiesUri(testEntity.getParentId(), testEntity.getTypeId() + 1, attributeNames));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+	}
+
+	@Test
+	public void testGetChildEntities_InvalidAttributeName_BadRequest() {
+		List<Object> attributeNames = new ArrayList<>();
+
+		for (int i = 0; i < 2; i++) {
+			attributeNames.add(testEntityTypeAttributes.get(i).getName() + "_invalid");
+		}
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getChildEntitiesUri(testEntity.getParentId(), testEntity.getTypeId(), attributeNames));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+	}
+
+	@Test
+	public void testGetEventTypesOfEntity() {
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getEventTypesOfEntityUri(testEntity.getId()));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		JsonArray eventTypes = jsonParser.parse(request.getResponse()).getAsJsonArray();
+		assertEquals(1, eventTypes.size());
+
+		JsonObject eventType = eventTypes.get(0).getAsJsonObject();
+		assertEquals(testEventType.getId(), eventType.get("Id").getAsLong());
+		assertEquals(testEventType.getName(), eventType.get("Name").getAsString());
+		assertEquals(1, eventType.get("NumberOfEvents").getAsLong());
+		assertEquals(testEventType.getTimeStampAttributeId(), eventType.get("TimestampAttributeId").getAsLong());
+
+	}
+
+	@Test
+	public void testGetEventTypesOfEntity_InvalidId_Success() {
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getEventTypesOfEntityUri(testEntity.getId() + 1));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
```

Comment:
The implementation does not check whether the given `Entity` really exists. Instead it just asks the database for all `Events` with the given `EntityId` as their owner. This list will be empty (because the `Entity` doesn't exist and therefore there can't be any `Events` belonging to the `Entity`).
But this does not change the response code due to consistence reasons.